libraries/abstractions/secure_sockets/lwip/iot_secure_sockets.c

```diff
@@ -809,6 +809,77 @@ int32_t SOCKETS_SetSockOpt( Socket_t xSocket,
 
             break;
 
+        case SOCKETS_SO_TCPKEEPALIVE:
+           {
+               int keep_alive = *( ( int * ) pvOptionValue );
+
+               if( ( keep_alive == 0 ) || ( keep_alive == 1 ) )
```

Comment:
Why is this validation necessary?  Why not just pass `pvOptionValue` directly to `lwip_setsockopt()` and let it validate the keepalive flag?

Reply:
The pvOptionValue is checked in 'lwip_setsockopt_impl()` API  if the value is zero the option is reset otherwise it is set.

I will remove the above check which is not required.

Reply:
done